pkg/endpoint/endpoint.go

```diff
@@ -681,6 +681,17 @@ func (e *Endpoint) applyOptsLocked(opts option.OptionMap) bool {
 	return changed
 }
 
+// ApplyOpts tries to lock endpoint, applies the given options to the
+// endpoint's options and returns true if there were any options changed.
+func (e *Endpoint) ApplyOpts(opts option.OptionMap) (bool, error) {
+	if err := e.lockAlive(); err != nil {
+		return false, err
+	}
+	defer e.unlock()
+	changed := e.applyOptsLocked(opts)
```

Comment:
After giving this another look I have one last question: is it safe to _override_ the endpoint opts? Could this trash any option which was set afterwards and was not part of the ones explicitly stored in `option.Config`?

For example: I was looking at the logic in `policy.go`:
https://github.com/cilium/cilium/blob/f66515219c21041f17d424cb8f2ca2b1b81c4687/pkg/endpoint/policy.go#L298-L312 

which in one specific case is responsible for setting an additional option (`Conntrack`)

Reply:
Thanks for the comment, if I understand you correctly, are you concerned about the specific case `option.Conntrack` not ensured when overriding endpoint opts? I've checked the code path, this is ensured right before regenerating bpf: https://github.com/cilium/cilium/blob/f66515219c21041f17d424cb8f2ca2b1b81c4687/pkg/endpoint/bpf.go#L831

Reply:
Thanks for confirming :+1: